src/apps/stable/features/playback/utils/mediaSessionSubscriber.ts

```diff
@@ -0,0 +1,178 @@
+        navigator.mediaSession.setActionHandler('play', this.onMediaSessionAction.bind(this));
+        navigator.mediaSession.setActionHandler('stop', this.onMediaSessionAction.bind(this));
+        navigator.mediaSession.setActionHandler('previoustrack', this.onMediaSessionAction.bind(this));
+        navigator.mediaSession.setActionHandler('nexttrack', this.onMediaSessionAction.bind(this));
+        navigator.mediaSession.setActionHandler('seekto', this.onMediaSessionAction.bind(this));
+        // iOS will only show next/prev track controls or seek controls
+        if (!browser.iOS) {
+            navigator.mediaSession.setActionHandler('seekbackward', this.onMediaSessionAction.bind(this));
+            navigator.mediaSession.setActionHandler('seekforward', this.onMediaSessionAction.bind(this));
+        }
+        /* eslint-enable compat/compat */
+    }
+
+    private onMediaSessionAction(details: MediaSessionActionDetails) {
+        switch (details.action) {
+            case 'pause':
+                return this.playbackManager.pause(this.player);
+            case 'play':
+                return this.playbackManager.unpause(this.player);
+            case 'stop':
+                return this.playbackManager.stop(this.player);
+            case 'seekbackward':
+                return this.playbackManager.rewind(this.player);
+            case 'seekforward':
+                return this.playbackManager.fastForward(this.player);
+            case 'seekto':
+                return this.playbackManager.seekMs((details.seekTime || 0) * MILLISECONDS_PER_SECOND, this.player);
+            case 'previoustrack':
+                return this.playbackManager.previousTrack(this.player);
+            case 'nexttrack':
+                return this.playbackManager.nextTrack(this.player);
+            default:
+                console.info('[MediaSessionSubscriber] Unhandled media session action', details);
+        }
+    }
+
+    private onMediaSessionUpdate(
+        { type: action }: Event,
+        state: PlayerState = this.playbackManager.getPlayerState(this.player)
+    ) {
+        const item = state.NowPlayingItem;
+
+        if (!item) {
+            console.debug('[MediaSessionSubscriber] no now playing item; resetting media session', state);
+            return resetMediaSession();
+        }
+
+        const isVideo = item.MediaType === MediaType.Video;
+        const isLocalPlayer = !!this.player?.isLocalPlayer;
+
+        // Local players do their own notifications
+        if (isLocalPlayer && isVideo) {
+            console.debug('[MediaSessionSubscriber] ignoring local player update');
+            return;
+        }
+
+        const album = item.Album || undefined;
+        const names = nowplayinghelper.getNowPlayingNames(item);
+        const artist = names[names.length - 1].text;
+        const title = names.length === 1 ? undefined : names[0].text;
```

Comment:
This condition is unnecessary? If you use nullish coalescing, text will be undefined if the index doesn't exist.

Reply:
Good call... I tend to forget about using nullish coalescing when accessing by index. :sweat_smile:

Reply:
I also see that you guys don't have https://www.typescriptlang.org/tsconfig/#noUncheckedIndexedAccess enabled, which in these cases it's a life saver

Reply:
Oh nice! I will enable that in a separate PR because it introduced 70 errors. :upside_down_face:

Reply:
Lucky you, I had a bunch of errors in Vue when I enabled it!